STM32F405RGTx/Core/Src/rtos_threads/measurements_thread.cpp

```diff
@@ -39,14 +44,43 @@ void MeasurementsThread::processData() {
         uint32_t val = (sum/ADC_DECIMATION_COEFF);
 
         switch (i) {
-            // TODO need to figure out which variables need to be loaded for each 'i'
-            // global_mc_data.pVa = ...;
-            // global_mc_data.pVb = ...;
-            // global_mc_data.pVc = ...;
+
+            // read and convert phase voltages and dc voltage
             case 0:
-                printf("Val = %lx\r\n", val);
+               g_mc_data.pVa = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 1:
+               g_mc_data.pVb = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 2:
+               g_mc_data.pVc = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+            case 3:
+               g_mc_data.dc_voltage = VOLTAGE_DIVIDER_CONVERSION( ADC_TO_VOLTAGE( val ) );
+
+            // read and convert phase currents
+            case 4:
+               g_mc_data.pIa = VOLTAGE_TO_CURRENT( ADC_TO_VOLTAGE(val) );
+            case 5:
+               g_mc_data.pIb = VOLTAGE_TO_CURRENT( ADC_TO_VOLTAGE(val) );
+            case 6:
+               g_mc_data.pIc = VOLTAGE_TO_CURRENT( ADC_TO_VOLTAGE(val) );
+
+            // read and convert adc temperatures
+            // NOTE: the ordering might change depending physical arrangement later on
+            case 7:
+            // Note: dc_cap_temp dne for Powerboard rev 2 but will for the next rev
+               g_mc_data.dc_cap_temp =  ADC_TO_TEMP_LUT[val];
+            case 8:
+               g_mc_data.fet_temps[0] = ADC_TO_TEMP_LUT[val];
+            case 9:
+               g_mc_data.fet_temps[1] = ADC_TO_TEMP_LUT[val];
+            case 10:
+               g_mc_data.fet_temps[2] = ADC_TO_TEMP_LUT[val];
 
         }
+
+        // update with functions once acceleration code is written
+        // g_mc_data.curr_accel =
+        // g_mc_data.curr_speed =
+        // g_mc_data.curr_pos =
     }
 }
 
```

Comment:
It won't let me write a comment on the actual function, but `MeasurementsThread::runMeasurements` needs to be implemented before this PR can be merged in. It needs to do the following in the while loop:

1. Process acceleration data
2. If the os flag 0x01 is set, process the data

Reply:
Does this mean this PR can't be merged until we merge the accelerometer code? (including the implemented filter)